validator/remote/src/main/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapter.java

```diff
@@ -199,54 +200,24 @@ private boolean currentEventStreamHasSameEndpoint(final RemoteValidatorApiChanne
         .equals(beaconNodeApi.getEndpoint());
   }
 
-  private void startBeaconNodeSyncingStatusQueryTask() {
-    final Cancellable currentTask = beaconNodeSyncingStatusQueryTask.get();
-    if (currentTask != null && !currentTask.isCancelled()) {
-      return;
-    }
-    final Cancellable task =
+  private void startBeaconNodeEventStreamReadinessCheckTask() {
+    beaconNodeEventStreamReadinessCheckTask =
         asyncRunner.runWithFixedDelay(
-            () ->
-                checkBeaconNodeIsReadyForEventStreaming(primaryBeaconNodeApi)
-                    .thenRun(
-                        () -> {
-                          if (maybeFailoverEventSource.isPresent()) {
-                            switchToPrimaryEventStream();
-                          }
-                        })
-                    .handleException(
-                        throwable -> {
-                          LOG.debug(
-                              "Primary Beacon Node is not ready for event streaming", throwable);
-                          switchToFailoverEventStreamIfNeeded();
-                        }),
-            beaconNodeEventStreamSyncingStatusQueryPeriod,
-            beaconNodeEventStreamSyncingStatusQueryPeriod,
+            () -> {
+              if (beaconNodeReadinessManager.isReady(primaryBeaconNodeApi)) {
+                if (maybeFailoverEventSource.isPresent()) {
+                  switchToPrimaryEventStream();
+                }
+              } else {
+                switchToFailoverEventStreamIfNeeded();
```

Comment:
looks like these don't call close on the old stream, which we should do i think.. they'll probably keep trying to reconnect? also `closeCurrentEventStream` is a little confusing, it may be close to time to refactor, as it implies that we've got multiple streams that may be open that we need to close... we don't want multiple streams open...

Reply:
> looks like these don't call close on the old stream, which we should do i think.. they'll probably keep trying to reconnect? also `closeCurrentEventStream` is a little confusing, it may be close to time to refactor, as it implies that we've got multiple streams that may be open that we need to close... we don't want multiple streams open...

They both close the current stream in the `switchToPrimaryEventStream` and `switchToFailoverEventStream` methods. I will look into refactoring.